scylla/src/transport/connection.rs

```diff
@@ -1550,6 +1553,22 @@ mod tests {
         }
     }
 
+    async fn check_schema_agreement(connections: &[&Connection]) -> Result<bool, QueryError> {
+        let handles = connections.iter().map(|c| c.fetch_schema_version());
+        let versions = try_join_all(handles).await?;
+
+        let local_version: Uuid = versions[0];
+        let in_agreement = versions.into_iter().all(|v| v == local_version);
+        Ok(in_agreement)
+    }
+
+    async fn await_schema_agreement(connections: &[&Connection]) -> Result<(), QueryError> {
+        while !check_schema_agreement(connections).await? {
+            tokio::time::sleep(Duration::from_millis(200)).await
+        }
+        Ok(())
+    }
```

Comment:
There is no need to reimplement a feature that the `Session` already supports. You should just create a session and then use it to send the CREATE TABLE query, the session will automatically wait for schema agreement afterwards. Don't duplicate logic without a good reason.

Reply:
So what you suggest here is using Session for all queries? Or using it for CREATE TABLE prologue only and then resorting to bare Connection? If so, should those two coexist independently or should I somehow use a connection from inside the Session?

Reply:
The test is supposed to test Connection::execute_all/query_all, so those methods need to use a bare connection (there aren't any equivalent methods in Session). Just use a Session to create a table. It's perfectly fine for both Session and bare Connection to coexist. If you wish, you can also drop the Session before creating the Connection, although I doubt it make any difference.